benchmark/fs/bench-mkdtempSync.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const tmpdir = require('../../test/common/tmpdir');
+tmpdir.refresh();
+
+const nonexistentDir = tmpdir.resolve('non-existent', 'foo', 'bar');
+const bench = common.createBenchmark(main, {
+  type: ['valid', 'invalid'],
+  n: [1e3],
+});
+
+function main({ n, type }) {
+  let prefix;
+
+  switch (type) {
+    case 'valid':
+      prefix = `${Date.now()}`;
+      break;
+    case 'invalid':
+      prefix = nonexistentDir;
+      break;
+    default:
+      new Error('Invalid type');
+  }
+
+  bench.start();
+  for (let i = 0; i < n; i++) {
+    try {
+      const folderPath = fs.mkdtempSync(prefix, { encoding: 'utf8' });
+      fs.rmdirSync(folderPath);
```

Comment:
I think we should remove the folder outside the benchmark (just append the index for every folder you create) as you also test here the delete for the valid case

Reply:
Pushed a fixup. Please review again

Reply:
Actually I reverted this change, because `mkdtempSync` adds a random suffix, and without storing the result on an array (which consumes a lot of memory), it's not possible to predict the outcome.